File: Domain/Activity.cs
using System.Security.AccessControl;

namespace Domain;

/// <summary>
/// Класс Активности,
/// наша доменная сущность
/// </summary>
public class Activity
{
    /// <summary>
    /// Индивидуальный идентификатор
    /// </summary>
    public Guid Id { get; set; }

    /// <summary>
    /// Название
    /// </summary>
    public string Title { get; set; }

    /// <summary>
    /// Дата
    /// </summary>
    public DateTime Date { get; set; }

    /// <summary>
    /// Описание
    /// </summary>
    public string Description { get; set; }
    
    /// <summary>
    /// Категория
    /// </summary>
    public string Category { get; set; }
    
    /// <summary>
    /// Город
    /// </summary>
    public string City { get; set; }
    
    /// <summary>
    /// Место проведения
    /// </summary>
    public string Venue { get; set; }
}

File: client-app/src/App/models/activity.ts
import {Profile} from "./profile";

export interface Activity {
    id: string
    title: string
    date: Date | null
    description: string
    category: string
    city: string
    venue: string
    hostUserName: string
    isCancelled: boolean
    isGoing: boolean
    isHost: boolean
    host?: Profile
    attendees: Profile[]
}

// dto
export class Activity implements Activity{
    constructor(init?: ActivityFormValues) {
        Object.assign(this,init)
    }
}

export class ActivityFormValues{
    id?: string = undefined
    title:string = ''
    category : string = ''
    description: string = ''
    city: string = ''
    venue: string = ''
    date: Date | null = null
    
    constructor(activity?: ActivityFormValues) {
        if (activity){
            this.id = activity.id
            this.title = activity.title
            this.category = activity.category
            this.description = activity.description
            this.city = activity.city
            this.venue = activity.venue
            this.date = activity.date
        }
    }
}
